### ISSUE TITLE
`cascade_invocations` to report only the first occurrence

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#3304]
**Describe the issue**
In order to provide a fix for `cascade_invocations` as requested in https://github.com/dart-lang/sdk/issues/47957, I believe it would require it to report only the first occurrence.

**To Reproduce**
```dart
class A {
  void m() {}
  int? x;
}

void f(A a) {
  a.m();
  a.x = 1; // <- here is triggered
  a.m();   // <- here also
}
```

**Expected behavior**
Only the first one in the chain should be linted, because for each lint the first invocation needs to be changed from `.` to `..`, so fixing the two lints at the same time would make conflict there.

**Comment by bwilkerson** [comment]
I don't think that it's required in order to provide a fix, but I do agree that it would be a better UX to flag the whole chain in a single lint. It would be unfortunate if a user were to fix the second occurrence of the lint and end up with
```
void f(A a) {
  a.m();
  a
    ..x = 1
    ..m();
}
```
I'm not sure whether the lint would fire in this case to indicate that the first invocation could be included in the cascade.

**Comment by asashour** [comment]
Actually the lint now fires in:

```dart
void f(A a) {
  a.m();
  a
    ..x = 1
    ..m();
}
```

Probably my question was more regarding the bulk fix, all of the lints together.

**Comment by bwilkerson** [comment]
Yes, for bulk fixes it's definitely more efficient to fix all of the occurrences together, but if the lint can also fire after the first fix is applied:
```
void f(A a) {
  a
    ..m()
    ..x = 1;
  a.m();
}
```
then we could still bulk fix it. The bulk fix implementation will apply the first edit, and if the second edit conflicted would drop the second edit. It will then iterate until there are no more diagnostics to fix (that part is fairly new).

But even ignoring the implementation of the bulk fix mechanism, I think it's a better UX to have one "These could all be cascaded" diagnostic than to have multiple "This could be cascaded" diagnostics.
